src/Avalonia.Visuals/Animation/Animators/GradientBrushAnimator.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System;
+using System.Collections.Generic;
+
+using Avalonia.Data;
+using Avalonia.Media;
+using Avalonia.Media.Immutable;
+
+#nullable enable
+
+namespace Avalonia.Animation.Animators
+{
+    /// <summary>
+    /// Animator that handles <see cref="SolidColorBrush"/> values. 
+    /// </summary>
+    public class IGradientBrushAnimator : Animator<IGradientBrush?>
```

Comment:
this class name can be confused with an Interface, should be just GradientBrushAnimator perhaps?

Reply:
I agree, but there is already ISolidColorBrushAnimator, so I named it IGradientBrushAnimator for consistency.
Also it makes sense to make both of them internal, since users are not meant to use it directly. Or even if they want, there is BrushBaseAnimator.

Reply:
But if we agree on renaming from IGradientBrushAnimator  to GradientBrushAnimator, I can do it in this PR. And create another PR to rename from ISolidColorBrushAnimator to SolidColorBrushAnimator. This second PR will be with breaking change, so won't be backported.